src/probnum/randprocs/_gaussian_process.py

```diff
@@ -63,45 +61,61 @@ class GaussianProcess(_random_process.RandomProcess[_InputType, _OutputType]):
 
     def __init__(
         self,
-        mean: Callable[[_InputType], _OutputType],
+        mean: _function.Function,
```

Comment:
I agree. The motivation here was that I did not want to assume general `Function`s to have scalar outputs by default. But since `Kernel`s have `shape=()` by default it seemed reasonable to have a `MeanFunction` doing the same thing